app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/image_import_workflow.rb

```diff
@@ -20,87 +51,119 @@ def post_execute
-      context[:task_id] = response.taskref.task_id
+      raise MiqException, 'HTTP response object is malformed, remote API error' if response&.taskref&.task_id.nil?
 
-      started_on = Time.now.utc
-      update!(:context => context, :started_on => started_on)
-      miq_task.update!(:started_on => started_on)
+      context[:task_id] = response.taskref.task_id
+      update!(:context => context)
+
+      msg = 'initiated OVA file importing from the cloud object storage'
+      signal = :post_execute_poll
+      status = 'ok'
+    rescue => e
+      msg = "failed to initiate OVA file importing from the cloud object storage: '#{trunc_err_msg(e.message)}'"
+      signal = :abort
+      status = 'error'
+      _log.error("OVA image import failure: '#{e.message}'")
     end
 
-    cleanup_git_repository
-
-    queue_signal(:post_execute_poll, 'importing image into PVS', 'running')
-  end
-
-  def post_execute_poll(*args)
-    msg, status = args
     set_status(msg, status)
+    queue_signal(signal, msg, status)
+  end
 
+  def post_execute_poll(*_args)
     ems = ExtManagementSystem.find(options[:ems_id])
 
     msg    = nil
     status = nil
     signal = nil
+    max_retries = 16
 
     ems.with_provider_connection(:service => 'PCloudTasksApi') do |api|
       response = api.pcloud_tasks_get(context[:task_id])
+      raise MiqException, 'HTTP response object is malformed, remote API error' if response&.status.nil?
 
       case response.status
       when 'capturing', 'downloading', 'creating', 'deleting', 'compressing', 'loading', 'started', 'uploading'
+        context[:retry] = 0
+        msg = "importing image into PVS, current state is: '#{response.status}'"
         signal = :post_execute_poll
-        msg = 'importing image into PVS, current state is: ' + response.status
-        status = 'running'
+        status = 'ok'
       when 'completed'
+        msg = 'importing image into image registry has completed'
         signal = :finish
-        msg = 'importing image into PVS has completed'
         status = 'ok'
       when 'failed'
-        signal = :error
-        msg = 'importing image into PVS has failed'
+        msg = "importing image into image registry has failed with an error: '#{trunc_err_msg(response.status)}'"
+        signal = :abort
+        status = 'error'
+        _log.error(response.status)
+      else
+        raise MiqException, "unexpected status - '#{response.status}'"
+      end
+    rescue => e
+      messsage = if e.kind_of?(IbmCloudPower::ApiError)
+                   suppress(StandardError) do
+                     JSON.parse(e.response_body)["description"] if e.response_body
+                   end
+                 end
+
+      messsage ||= e.message
+
+      retr = context[:retry].to_i
+      if retr < max_retries
+        msg = "failed to retrieve status for a cloud-task: '#{trunc_err_msg(messsage)}'; try number: #{retr + 1} of #{max_retries}"
+        signal = :post_execute_poll
```

Comment:
If you're going to loop on a retry you probably want to add a queue delay so it doesn't just spin, e.g. add `:deliver_on => Time.now.utc + 1.minute` to the `queue_signal` method opts